### ISSUE TITLE
AQL Arithmetic multiplication got unexpected result

### ISSUE THREAD

**Issue by greatwar** [arangodb/arangodb#2392]
## my environment running ArangoDB
I'm using the latest ArangoDB of the respective release series:
- [ ] 2.8
- [ ] 3.0
-  3.1
-  3.2 pre-releases: alpha2
- [ ] self-compiled devel branch

On this operating system:
- [ ] DCOS on
  - [ ] AWS
  - [ ] Azure
  - [ ] own infrastructure
- [ ] Linux 
  - [ ] Debian .deb
  -  Ubuntu .deb
  - [ ] SUSE   .rpm
  - [ ] RedHat .rpm
  - [ ] Fedora .rpm
  - [ ] Gentoo
  - [ ] docker - official docker library
  - [ ] other:
- [ ] Windows, version: 
- [ ] MacOS, version:

### this is an AQL-related issue:
[ ] I'm using graph features

I'm issuing AQL via:
-  web interface with this browser:     running on this OS:
-  arangosh
- [ ] this Driver:

I've run `db._explain("<my aql query>")` and it didn't shed more light on this.
The AQL query in question is:
```
let it={a:4,b:4,c:4} return {a:it.a * 3,b:it.b * 3,c:it.c * 3}
```

expect: {"a":12,"b":12,"c":12} but got {"a":12,"b":**0**,"c":**0**}

_explain:
```
db._explain("let it={a:4,b:4,c:4} return {a:it.a * 3,b:it.b * 3,c:it.c * 3}  ")
Query string:
 let it={a:4,b:4,c:4} return {a:it.a * 3,b:it.b * 3,c:it.c * 3}  

Execution plan:
 Id   NodeType          Est.   Comment
  1   SingletonNode        1   * ROOT
  3   CalculationNode      1     - LET #1 = { "a" : 12, "b" : 0, "c" : 0 }   /* json expression */   /* const assignment */
  4   ReturnNode           1     - RETURN #1

Indexes used:
 none

Optimization rules applied:
 Id   RuleName
  1   move-calculations-up
  2   remove-unnecessary-calculations
```

The issue can be reproduced without any dataset

**Comment by jsteemann** [comment]
Fixed in 3.1.16.

**Comment by dothebart** [comment]
3.1.16 is now available for Download - closing.
